tools/UbuntuBootstrapper.php

```diff
@@ -0,0 +1,204 @@
+     *
+     * @return void
+     */
+    public function __construct(array $args = array())
+    {
+        parent::__construct($args);
+        $required_php = self::PHP_MAJOR_VERSION_REQUIRED
+            . '.'
+            . self::PHP_MINOR_VERSION_REQUIRED;
+
+        // Dependencies last updated for version: 20.2.0
+        // This list should consist only of packages that can be installed
+        // via apt on Ubuntu environments and must not include libraries that
+        // should be installed via tools such as npm and composer.
+        $this->requirements = array(
+                               "curl",
+                               "wget",
+                               "zip",
+                               "unzip",
+                               "php-json",
+                               "make",
+                               "software-properties-common",
+                               "php-ast",
+                               "php$required_php",
+                               "php$required_php-mysql",
+                               "php$required_php-xml",
+                               "php$required_php-json",
+                               "php$required_php-mbstring",
+                               "php$required_php-gd",
+                               "libapache2-mod-php$required_php",
+                              );
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return bool Whether the requirement is met.
+     */
+    public function phpRequirementSatisfied(): bool
+    {
+        return PHP_MAJOR_VERSION >= self::PHP_MAJOR_VERSION_REQUIRED
+            && PHP_MINOR_VERSION >= self::PHP_MINOR_VERSION_REQUIRED;
+    }
+
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return bool Whether the requirement is met.
+     */
+    public function apacheRequirementSatisfied(): bool
+    {
+        // Get string representation of apache version number.
+        // This command yields e.g. Apache/2.4.34
+        $cmdOutput = shell_exec(
+            "apache2 -v | " .
+            "head -n 1 | " .
+            "cut -d ' ' -f 3"
+        );
+        if (is_null($cmdOutput)) {
+            // Apache2 is not installed.
+            return false;
+        }
+        $apacheParts   = explode(
+            '/',
+            $cmdOutput
+        );
+        $apacheVersion = end($apacheParts);
+
+        // Look for the string "$major.$minor" in info string. Also match on versions
+        // higher than minor version because we want AT LEAST that version.
+        $pattern = sprintf(
+            "/%s\.[%s-9].[0-9]/",
+            self::APACHE_MAJOR_VERSION_REQUIRED,
+            self::APACHE_MINOR_VERSION_REQUIRED
+        );
+
+        // When preg_match returns 0 it means no match was found.
+        return preg_match($pattern, $apacheVersion) !== 0;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return bool Whether the requirement is met.
+     */
+    public function nodeJSRequirementSatisfied(): bool
```

Comment:
This function is never called.

Reply:
Thanks, the latest commit should fix this.